solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -413,35 +409,30 @@ private ElevationProvider loadElevationProvider(SolrCore core) throws Exception
    * @throws RuntimeException If the config does not provide an XML content of the expected format
    *     (either {@link RuntimeException} or {@link org.apache.solr.common.SolrException}).
    */
-  protected ElevationProvider loadElevationProvider(XmlConfigFile config) {
+  protected ElevationProvider loadElevationProvider(Document doc) {
     Map<ElevatingQuery, ElevationBuilder> elevationBuilderMap = new LinkedHashMap<>();
-    XPath xpath = XPathFactory.newInstance().newXPath();
```

Comment:
why did we move away from the XPath model? because it's slow or for some other reason?

Reply:
because the need for which it is used is simple and it adds one more dependence when we can get rid of it without complicating the code.